src/Database/Log/LoggingStatement.php

```diff
@@ -51,40 +65,76 @@ class LoggingStatement extends StatementDecorator
      */
     public function execute(?array $params = null): bool
     {
-        $t = microtime(true);
-        $query = new LoggedQuery();
+        $this->startTime = microtime(true);
+
+        $this->loggedQuery = new LoggedQuery();
+        $this->loggedQuery->params = $params ?: $this->_compiledParams;
```

Comment:
We could make it a constructor argument but since all properties of `LoggedQuery` are public I don't see much benefit. Maybe in future we can make them protected and update the public API accordingly.